File: CS-Examples/19_PivotTables/SetFormatOptions.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace SetFormatOptions
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook class
            Dim workbook As New Workbook()

            ' Load the Excel file into the Workbook object
            workbook.LoadFromFile("..\..\..\..\..\..\Data\PivotTableExample.xlsx")

            ' Access the specific worksheet by name and assign it to a Worksheet object
            Dim sheet As Worksheet = workbook.Worksheets("PivotTable")

            ' Retrieve the first PivotTable on the worksheet and cast it to XlsPivotTable object
            Dim pt As XlsPivotTable = TryCast(sheet.PivotTables(0), XlsPivotTable)

            ' Enable automatic formatting for the PivotTable
            pt.Options.IsAutoFormat = True

            ' Show grand totals for rows in the PivotTable
            pt.ShowRowGrand = True

            ' Show grand totals for columns in the PivotTable
            pt.ShowColumnGrand = True

            ' Set the option to display a specified string for null values in the PivotTable
            pt.DisplayNullString = True
            pt.NullString = "null"

            ' Set the page field order for multiple page fields in the PivotTable
            pt.PageFieldOrder = PagesOrderType.DownThenOver

            ' Specify the output file name after saving the modified Workbook
            Dim result As String = "SetFormatOptions_result.xlsx"
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/19_PivotTables/SetFormatOptions.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace SetFormatOptions
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load an excel file including pivot table
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\PivotTableExample.xlsx");

            // Get the sheet where the pivot table is located
            Worksheet sheet = workbook.Worksheets["PivotTable"];

            // Access the first pivot table in the sheet
            XlsPivotTable pt = sheet.PivotTables[0] as XlsPivotTable;

            // Enable automatic formatting for the pivot table report
            pt.Options.IsAutoFormat = true;

            // Show grand totals for rows in the pivot table report
            pt.ShowRowGrand = true;

            // Show grand totals for columns in the pivot table report
            pt.ShowColumnGrand = true;

            // Display a custom string in cells that contain null values
            pt.DisplayNullString = true;
            pt.NullString = "null";

            // Set the layout of the pivot table report
            pt.PageFieldOrder = PagesOrderType.DownThenOver;

            // Specify the filename for the resulting workbook
            string result = "SetFormatOptions_result.xlsx";

            // Save the modified workbook to a file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
